src/jit/emitxarch.cpp

```diff
@@ -267,7 +267,9 @@ bool emitter::Is4ByteSSE4Instruction(instruction ins)
 bool emitter::TakesVexPrefix(instruction ins)
 {
     // special case vzeroupper as it requires 2-byte VEX prefix
-    if (ins == INS_vzeroupper)
+    // special case sfence as it never takes a VEX prefix
+    if ((ins == INS_vzeroupper) || (ins == INS_sfence) || (ins == INS_prefetcht0) || (ins == INS_prefetcht1) ||
+        (ins == INS_prefetcht2) || (ins == INS_prefetchnta))
```

Comment:
As I know, these instructions do not have VEX-encoding.

Reply:
Yes, this ensures they are not given a VEX prefix.

Reply:
Oh, sorry, I misread.

Reply:
Maybe you could update the comment

Reply:
Fixed.